git_machete/tests/functional/test_machete.py

```diff
@@ -36,110 +37,128 @@ def __init__(self) -> None:
         self.json_data: Union[str, bytes] = data
         self.return_data: Optional[Union[List[Dict[str, Any]], Dict[str, Any]]] = None
         self.headers: Dict[str, str] = headers
+        self.handle_method(method)
+        return ResponseWrapper(self.return_data)
+
+    def handle_method(self, method: str) -> None:
         if method == "GET":
             self.handle_get()
         elif method == "PATCH":
             self.handle_patch()
         elif method == "POST":
             self.handle_post()
 
-    def read(self) -> Union[str, bytes]:
-        return json.dumps(self.return_data).encode()
-
     def handle_get(self) -> None:
-        if 'pulls' in self.url:
-            if '?head' in self.url:
-                head: str = self.url[self.url.rfind(':') + 1:]
-                for pr in GitAPIState.pulls:
-                    if pr['head']['ref'] == head:
+        if self.parsed_url.path.endswith('pulls'):
+            full_head_name: Optional[List[str]] = self.parsed_query.get('head')
+            if full_head_name:
+                head: str = full_head_name[0]
+                head_name: str = head[head.rfind(':') + 1:]
+                for pr in self.git_api_state.pulls:
+                    if pr['head']['ref'] == head_name:
                         self.return_data = [pr]
-                        self.status_code = 200
+                        self.status_code = HTTPStatus.OK
                         return
                 self.return_data = []
-                self.status_code = 404
+                self.status_code = HTTPStatus.NOT_FOUND
                 return
             else:
-                self.status_code = 200
-                self.return_data = GitAPIState.pulls
+                self.status_code = HTTPStatus.OK
+                self.return_data = self.git_api_state.pulls
                 return
-        elif 'user' in self.url:
-            self.status_code = 200
-            self.return_data = GitAPIState.user
+        elif self.parsed_url.path.endswith('user'):
+            self.status_code = HTTPStatus.OK
+            self.return_data = self.git_api_state.user
             return
 
     def handle_patch(self) -> None:
-        if 'issues' in self.url:
+        if 'issues' in self.parsed_url.path:
             self.update_issue()
-        elif 'pulls' in self.url:
+        elif 'pulls' in self.parsed_url.path:
             self.update_pull()
 
     def handle_post(self) -> None:
-        if 'issues' in self.url:
-            if self.url.endswith('issues'):
-                self.update_issue(new=True)
-        elif 'pulls' in self.url:
-            if self.url.endswith('pulls'):
-                self.update_pull(new=True)
+        assert not self.parsed_query
+        if self.parsed_url.path.endswith('issues'):
+            self.update_issue(new=True)
+        elif self.parsed_url.path.endswith('pulls'):
+            self.update_pull(new=True)
 
     def update_pull(self, new: bool = False) -> None:
```

Comment:
As discussed in one of the past PRs (the one about refactoring shell code), I'd avoid making switches based on the flag value. Pls extract separate `create_pull_request` and `create_issue` methods and delegate the shared parts of logic to a third method if needed